File: backend/src/main/java/com/mysite/knitly/domain/design/repository/DesignRepository.java
package com.mysite.knitly.domain.design.repository;

import com.mysite.knitly.domain.design.entity.Design;
import com.mysite.knitly.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface DesignRepository extends JpaRepository<Design,Long> {
    List<Design> findByUser(User user);
}

File: backend/src/main/java/com/mysite/knitly/domain/design/repository/DesignRepository.kt
package com.mysite.knitly.domain.design.repository

import com.mysite.knitly.domain.design.entity.Design
import com.mysite.knitly.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface DesignRepository : JpaRepository<Design, Long> {
    fun findByUser(user: User): List<Design>
}
